Pipeline:filters:metadata: updating for style (#1675)
fluent/fluent-bit-docs: pipeline/filters/aws-metadata.md

# AWS Metadata

The _AWS Filter_ enriches logs with AWS Metadata. The plugin adds the EC2 instance ID and availability zone to log records. To use this plugin, you must be running in EC2 and have the [instance metadata service enabled](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html).

## Configuration parameters

The plugin supports the following configuration parameters:

| Key | Description | Default |
| :--- | :--- | :--- |

> | `imds_version` | Specify which version of the instance metadata service to use. Valid values are 'v1' or 'v2'. | v2 |

Reviewer: NB: Input formatting

```suggestion
| `imds_version` | Specify which version of the instance metadata service to use. Valid values are `v1` and `v2`. | `v2` |
```

Author: ah shoot those are hard to spot, thanks

> | `az` | The [availability zone](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html); for example, `us-east-1a`. | `true` |

Reviewer: NB: Phrasing tweak

```suggestion
| `az` | The [availability zone](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html), such as `us-east-1a`. | `true` |
```

| `ec2_instance_id` | The EC2 instance ID. | `true` |
| `ec2_instance_type` | The EC2 instance type. | `false` |
| `private_ip` | The EC2 instance private IP. | `false` |

> | `ami_id` | The EC2 instance image id. | `false` |

Reviewer: NB: ID case

```suggestion
| `ami_id` | The EC2 instance image ID. | `false` |
```

> | `account_id` | The account ID for current EC2 instance. | `false` |
> | `hostname` | The hostname for current EC2 instance. | `false` |
> | `vpc_id` | The VPC ID for current EC2 instance. | `false` |

Reviewer: NB: Missing words

```suggestion
| `account_id` | The account ID for the current EC2 instance. | `false` |
| `hostname` | The hostname for the current EC2 instance. | `false` |
| `vpc_id` | The VPC ID for the current EC2 instance. | `false` |
```

> | `tags_enabled` | Specifies if should attach EC2 instance tags. EC2 instance must have the [instance-metadata-tags](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ec2/modify-instance-metadata-options.html) option enabled, which is disabled by default. | `false` |

Reviewer: NB: Grammar fix suggestion ("specifies if should") and formatting

```suggestion
| `tags_enabled` | Specifies whether to attach EC2 instance tags. The EC2 instance must have the [`instance-metadata-tags`](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ec2/modify-instance-metadata-options.html) option enabled, which is disabled by default. | `false` |
```

> | `tags_include` | Defines list of specific EC2 tag keys to inject into the logs. Tag keys must be comma-separated `,`. Tags which aren't present in this list will be ignored. Example: `Name,tag1,tag2`. | _none_ |
> | `tags_exclude` | Defines list of specific EC2 tag keys not to inject into the logs. Tag keys must be comma-separated `,`. Tags which aren't present in this list will be injected into the logs. If both `tags_include` and `tags_exclude` are specified, configuration is invalid and plugin fails. Example: `Name,tag1,tag2` | _none_ |

Reviewer: NB: Missing words, put illustrative comma in parentheses, tighten phrasing

```suggestion
| `tags_include` | Defines a list of specific EC2 tag keys to inject into the logs. Tag keys must be comma-separated (`,`). Tags not included in this list will be ignored. Example: `Name,tag1,tag2`. | _none_ |
| `tags_exclude` | Defines a list of specific EC2 tag keys not to inject into the logs. Tag keys must be comma-separated (`,`). Tags not included in this list will be injected into the logs. If both `tags_include` and `tags_exclude` are specified, the configuration is invalid and the plugin fails. Example: `Name,tag1,tag2` | _none_ |
```

    Match *
```

## EC2 tags

> EC2 Tags let you to label and organize your EC2 instances by creating custom-defined key-value pairs. These tags are commonly utilized for resource management, cost allocation, and automation. Including them in the Fluent Bit generated logs is almost essential.

Reviewer: Fix typo introduced by edit, avoid "utilized" for "used", and hyphenate compound term:

```suggestion
EC2 Tags let you label and organize your EC2 instances by creating custom-defined key-value pairs. These tags are commonly used for resource management, cost allocation, and automation. Including them in the Fluent Bit-generated logs is almost essential.
```

### Example

#### `tags_include`

Assume the EC2 instance has many tags, some of which have lengthy values that are irrelevant to the logs you want to collect. Only two tags, `department` and `project`, are valuable for your purpose. The following configuration reflects this requirement:

```python
[FILTER]
    Name aws
    Match *
    tags_enabled true
    tags_include department,project
```

If you run Fluent Bit logs might look like the following:

```text
{"log"=>"fluentbit is awesome", "az"=>"us-east-1a", "ec2_instance_id"=>"i-0e66fc7f9809d7168", "department"=>"it", "project"=>"fluentbit"}
```

#### `tags_exclude`

> Suppose the EC2 instance has three tags: `Name:fluent-bit-docs-example`, `project:fluentbit`, and `department:it`. In this example,  the `department` tag is redundant and will be excluded. All of the projects belong to the `it` department, and you don't want to waste storage space on redundant labels.

Reviewer: NB: Remove extraneous space added in edit

```suggestion
Suppose the EC2 instance has three tags: `Name:fluent-bit-docs-example`, `project:fluentbit`, and `department:it`. In this example, the `department` tag is redundant and will be excluded. All of the projects belong to the `it` department, and you don't want to waste storage space on redundant labels.
```
